pkg/controller/machineset.go

```diff
@@ -349,6 +357,12 @@ func (c *controller) manageReplicas(allMachines []*v1alpha1.Machine, machineSet
 		}
 
 		diff *= -1
+
+		// clamp machine creation to the count of recently deleted stale machines, so that we don't overshoot
+		if staleMachinesLeft > 0 && diff > deletedStaleMachines {
+			diff = deletedStaleMachines
```

Comment:
Hm, this seems to be breaking the scale-up situation.
Eg. active machines are 10, desired is 13, and stale machines are 4, and thresold: 50% .
Then I would want to create 2+3=5 machines, and not 2.